src/questionnaireitem_system_focus.js

```diff
@@ -0,0 +1,58 @@
+/**
+A QuestionnaireItemSystem that stores how often and how long the focus was moved away from the page.
+
+Reports as array in milliseconds how long the focus was lost each time.
+
+@class QuestionnaireItemSystemFocus
+@augments UIElement
+@augments UIElementInteractive
+@augments QuestionnaireItem
+@augments QuestionnaireItemSystem
+*/
+function QuestionnaireItemSystemFocus() {
+    QuestionnaireItemSystem.call(this, null, "Focus", false);
+
+    this.focusLossDurations = [];
+    this.answer = this.focusLossDurations;
+    this.timeOfLastFocusLoss = new Date().getTime();
+    this.inFocus = null;
```

Comment:
Why null?
What does it mean?

Reply:
It is possible to open the survey in a new tab. Then the website gains focus first. To measure how long the duration was before gaining focus, which is also a focus loss, I need to set inFocus to null and also save time there.
I consider moving the last two lines into the createUI() method. Because the element could be created a long time before actually being displayed.